### ISSUE TITLE
Consider making some optional members also nullable

### ISSUE THREAD

**Issue by kainino0x** [gpuweb/gpuweb#1400]
In https://crbug.com/1153201 a user was surprised to find that `resolveTarget: null` was invalid even though omitting it (or specifying `resolveTarget: undefined`) is valid.

As a small ergonomic improvement we could consider making some optional members also `nullable` to make that work.
There may be a tiny bindings performance hit to doing this.

This is quite minor and would be a non-breaking change, so it is definitely post-MVP.

**Comment by Kangz** [comment]
I think we should discuss this for V1 as polish for the API.

**Comment by kainino0x** [comment]
I would like to do this. I started a thread with our Chrome spec mentors to ask about whether this is conventional.

**Comment by kainino0x** [comment]
I got negative feedback from 2 mentors on having our API allow both null and undefined:

> it's most consistent with the web platform to have undefined trigger default [behavior].
> I'd encourage sticking with the undefined-only behavior and using this as an opportunity to help users understand better how the web platform (and JavaScript APIs generally, e.g. ones like `function f(x = "5") { console.log(x); }`) work.

(That is, when you call `f(undefined)`, `x` is `"5"`, but when you call `f(null)`, `x` is `null`. I had thought of this argument in the past but forgotten. Perhaps better illustrated by [this example](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABMAFHADlAzgLkQb0QA8B+PLKAJxjAHNEAfRMEAG1cQF8BKAgKESDEEBBQLFEAXkQAiAKwyuUxBmwBuAUJFgscVgFMAdKzi0URbhs58+qQkTzgAJvuA19TrpcQB6H4nRqMGxZBVsUezwWdi81X39AmhCZaNYZIA) which uses a dictionary instead of a function argument.)

> null only makes sense in cases where we need to be able to distinguish between "nothing" (and the user explicitly specified that) and "nothing" (maybe it's just the default). That should be relatively rare and probably hints at an API design problem.

Both of these responses reflect the understanding we went into the API design with - we should be consistent with JS language features, and there shouldn't be two ways to do the same thing - so I'm actually glad to have it reinforced.

The conversation made me realize the real problem is probably just Chrome's error message being misleading (Firefox has the same problem). I filed https://crbug.com/1285483.

----

So I propose rejecting this proposal unless someone still really wants it. However there are 2 related things we should consider:
- Should we make GPUObjectBase.label (which is nullable but can't be `undefined) match GPUObjectDescriptorBase.label (which is optional but can't be `null`)? I think this would just look like `USVString?` -> `(USVString | undefined)`.
- Are any places in the API we should use nullable instead of optional? (Force users to specify null.) I am pretty sure we would have considered this already if there were, but I didn't check.
